phpunit/blocks/class-wp-navigation-block-renderer-test.php

```diff
@@ -63,6 +63,76 @@ public function test_gutenberg_get_markup_for_inner_block_site_title() {
 		$this->assertEquals( $expected, $result );
 	}
 
+	/**
+	 * Test that the heading block isn't wrapped in a list item to preserve accessible markup
+	 *
+	 * @group navigation-renderer
+	 *
+	 * @covers WP_Navigation_Block_Renderer::get_markup_for_inner_block
+	 */
+	public function test_gutenberg_get_markup_for_inner_block_heading() {
+
+		// We are testing the site title block because we manually add list items around it.
+		$parsed_blocks = parse_blocks(
+			'<!-- wp:heading --><h2 class="wp-block-heading">Hello World</h2><!-- /wp:heading -->'
+		);
```

Comment:
This is a bit of an ask, but I wonder if we could instead register a test block so that it's obvious we're just testing a random block and not the specifics of the core/heading block?

Reply:
Updated :)